main.go

```diff
@@ -951,3 +900,262 @@ func addControllers(
 		os.Exit(1)
 	}
 }
+
+// gatekeeperInstallationChecker starts or stops the gatekeeper-constraint-status-sync controller if Gatekeeper's
+// installation status changes. The controller must be off when Gatekeeper is not installed.
+func gatekeeperInstallationChecker(
+	ctx context.Context, managedCfg *rest.Config, dynamicClient dynamic.Interface, mgrOptions manager.Options,
+) error {
```

Comment:
I think the cases where this function returns an error are just:
1. if it can't get a free local address
2. if the initial attempt to make the retry watcher fails.

So, if case 1 moves out of the function, and case 2 could be handled like non-initial retry watcher creation failures are handled (by retrying, I think?) then this function would not return an error.

The reason I'm thinking in that direction is that this function ends with a `go func() ...`, so I feel like most of its "work" is asynchronous. But the name of the function, and the calling convention doesn't make that obvious. If we called it as `go gkInstallChecker(...)`, it might be more clear what it is doing? Plus it would save some indents at the end.

I don't know - maybe it wouldn't be better. Just something to consider